authentication/api/main.py

```diff
@@ -189,6 +150,16 @@ async def token(
     """
     if x_amzn_mtls_clientcert is None:
         raise HTTPException(status_code=401, detail="No client certificate provided")
+    try:
+        auth.require_role(
+            "https://registry.core.ib1.org/scheme/perseus/role/carbon-accounting",
+            x_amzn_mtls_clientcert,
+        )
+    except auth.CertificateError as e:
```

Comment:
When I wrote the original of this, I was hoping it could just be a simple statement. Catching and raising another feels ugly. Rename to `require_role_for_request()` and just throw the `HTTPException`?

Reply:
Starting to think of its use as a library, where the function could be used in different contexts - the library throws its own descriptive errors, you can deal with it as needed - in this case we're catching any kind of CertificateError and returning an http error